ezo/factory.go

```diff
@@ -0,0 +1,95 @@
+package ezo
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/reef-pi/hal"
+	"github.com/reef-pi/rpi/i2c"
+)
+
+type factory struct {
+	meta       hal.Metadata
+	parameters []hal.ConfigParameter
+}
+
+var ezoFactory *factory
+var once sync.Once
+
+// Factory returns a singleton EZO Driver factory
+func Factory() hal.DriverFactory {
+
+	once.Do(func() {
+		ezoFactory = &factory{
+			meta: hal.Metadata{
+				Name:         _ezoName,
+				Description:  "Atlas Scientific EZO board for pH sensor",
+				Capabilities: []hal.Capability{hal.AnalogInput},
+			},
+			parameters: []hal.ConfigParameter{
+				{
+					Name:    "Address",
+					Type:    hal.Integer,
+					Order:   0,
+					Default: 68,
+				},
+			},
+		}
+	})
+
+	return ezoFactory
+}
+
+func (f *factory) Metadata() hal.Metadata {
+	return f.meta
+}
+
+//Implement hal.Driver interface
+func (f *factory) GetParameters() []hal.ConfigParameter {
+	return f.parameters
+}
+
+func (f *factory) ValidateParameters(parameters map[string]interface{}) (bool, map[string][]string) {
+
+	var failures = make(map[string][]string)
+
+	if address, ok := parameters["Address"]; ok {
```

Comment:
Would be nice to declare "Address" as a constant

Reply:
That makes sense.  I can refactor and update parameter names to constants.